src/main/java/com/adobe/target/edge/client/ondevice/collator/CustomParamsCollator.java

```diff
@@ -35,4 +36,29 @@ private void addAllParameters(Map<String, Object> custom, RequestDetails details
       }
     }
   }
+
+  private Map<String, Object> createNestedParametersFromDots(Map<String, Object> custom) {
+    Map<String, Object> result = new HashMap<>();
+    custom.forEach(
+        (key, value) -> {
+          if (!key.contains(".")
```

Comment:
Let's invert IF condition and return early, so we could remove ELSE and `addNestedKeyToParameters()` nesting